src/ibex_bluesky_core/devices/dae/dae_period_settings.py

```diff
@@ -0,0 +1,132 @@
+"""ophyd-async devices and utilities for the DAE hardware period settings."""
+
+import xml.etree.ElementTree as ET
+from dataclasses import dataclass
+from enum import Enum
+from typing import List
+from xml.etree.ElementTree import tostring
+
+from bluesky.protocols import Locatable, Location
+from ophyd_async.core import AsyncStatus, Device, SignalRW
+
+from ibex_bluesky_core.devices import (
+    isis_epics_signal_rw,
+)
+from ibex_bluesky_core.devices.dae import (
+    convert_xml_to_names_and_values,
+    get_all_elements_in_xml_with_child_called_name,
+    set_value_in_dae_xml,
+)
+
+OUTPUT_DELAY = "Output Delay (us)"
+PERIOD_SEQUENCES = "Hardware Period Sequences"
+PERIOD_FILE = "Period File"
+PERIOD_SETUP_SOURCE = "Period Setup Source"
+PERIOD_TYPE = "Period Type"
+PERIODS_SOFT_NUM = "Number Of Software Periods"
+
+
+class PeriodType(Enum):
+    """General period type."""
+
+    SOFTWARE = 0
+    HARDWARE_DAE = 1
+    HARDWARE_EXTERNAL = 2
+
+
+class PeriodSource(Enum):
+    """The period setup source, whether to use parameters or file."""
+
+    PARAMETERS = 0
+    FILE = 1
+
+
+@dataclass
+class SinglePeriodSettings:
+    """Dataclass for the settings on a single period."""
+
+    type: int | None = None
+    frames: int | None = None
+    output: int | None = None
+    label: str | None = None
+
+
+@dataclass
+class DaePeriodSettingsData:
+    """Dataclass for the hardware period settings."""
+
+    periods_settings: List[SinglePeriodSettings]
```

Comment:
I guess this means we _always_ have to specify periods_settings - is there a way around that? It feels like we should be able to accept `None` for this?

Reply:
yes, https://github.com/ISISComputingGroup/ibex_bluesky_core/pull/27/commits/22fc9fc8c7babe2d1a8e983a2217bac8d1097893